### PR TITLE
docs: autogenerate cmd docs; add more `install --help` text; update `customizing-install` for config files

### PR DESCRIPTION
**Pull request by bgilbert**
It turns out that clap distinguishes `-h` and `--help`, and we can put additional text in `--help` without affecting the compact `-h` output.  Since it's easier not to maintain command docs in two places, switch to autogenerating `docs/cmd/`.

As a result, we need to move the config file docs somewhere else.  Move them to `customizing-install` and restructure that page for the `installer.d` config file flow.  Add tests to ensure every `install` long option is documented in the example config file.

Also add some long-help docs for the `install` subcommand.

### CODE REVIEW

**Comment by bgilbert** [comment]
Rendered at https://bgilbert.github.io/coreos-installer/.

File: docs/_cmd.sh [hunk 1]
```diff
@@ -30,9 +30,14 @@ EOF
...
 \`\`\`
 EOF
+        # Run coreos-installer in a PTY and override the terminal width to
+        # the one we want, regardless of the caller's terminal.  More than
+        # 100 characters will cause the code block to scroll.
+        # Drop CR characters added by `script`.
         # Drop first line with incorrectly hyphenated command name and version
         # Fix trailing whitespace
-        "${prog}" $* --help | \
+        script -qc "stty cols 95 rows 24; ${prog} $* --help" /dev/null | \
```
[the remark this answers is not in the document]
Comment (bgilbert) [reply docs/_cmd.sh:39]: Yeah, we're only using it here for its PTY side effect.

File: src/cmdline.rs [hunk 2]
```diff
@@ -165,12 +165,24 @@ pub enum PxeIgnitionCmd {
...
     /// YAML config file with install options
+    ///
+    /// Load additional config options from the specified YAML config file.
+    /// Later config files override earlier ones, and command-line options
+    /// override config files.
+    ///
+    /// Config file keys are long option names without the leading "--".
+    /// Values are strings for non-repeatable options, arrays of strings for
+    /// repeatable options, and "true" for flags.  The destination device
+    /// can be specified with the "dest-device" key.
```
[the remark this answers is not in the document]
Comment (bgilbert) [reply src/cmdline.rs:176]: It'd be nice, but the URL is too long to render cleanly.  Going to leave it out for now.

**Comment by bgilbert** [comment]
Updated!

**Review by jlebon — APPROVED** [review]

**Comment by cgwalters** [comment]
Really nice work on this!  Would be good to standardize this flow across other repos I think.

**Pull request merged**